rclcpp/include/rclcpp/executor.hpp

```diff
@@ -231,6 +231,55 @@ class Executor
     return FutureReturnCode::INTERRUPTED;
   }
 
+  template<typename ResponseT, typename TimeRepT = int64_t, typename TimeT = std::milli>
+  FutureReturnCode
+  spin_until_future_complete(
+    const std::future<ResponseT> & future,
+    std::chrono::duration<TimeRepT, TimeT> timeout = std::chrono::duration<TimeRepT, TimeT>(-1))
```

Comment:
What I see, is that this is exactly a repetition of the template method above, but not just taking the "future response" as a template parameter, instead of taking ResponseT. i.e. update the current template definition to:

```cpp
  template<typename FutureResponseT, typename TimeRepT = int64_t, typename TimeT = std::milli>
  FutureReturnCode
  spin_until_future_complete(
    const FutureResponseT && future,
    std::chrono::duration<TimeRepT, TimeT> timeout = std::chrono::duration<TimeRepT, TimeT>(-1))
```

Reply:
This comment applies for all the other methods too